### ISSUE TITLE
How to deal with the situation when the number of inner-loop steps at test time is different from training time?

### ISSUE THREAD

**Issue by Big-Wuu** [AntreasAntoniou/HowToTrainYourMAMLPytorch#28]
Dear authors,
Your ideas to deal with the instability of MAML is great and elegant. But there is a simple but vital question with Per-step BN and Per-step LR, that's can they deal with the situation when the number of inner-loop steps at test time is different from training time? e.g. I only update 5 times during training but I want to update more times at test. It seems can't because we don't have the statistics out of range.

**Comment by AntreasAntoniou** [comment]
The best way to do this would be to increase the steps at training time to
reflect the setup you want for testing time. If you must train for less
steps than you test, then try the following:

1. Learn a learning rate for each layer for all the steps (i.e. having only
1 learning rate for each layer instead of 5).
2. Replace batch norm with a normal, inner loop adaptable batch norm.

Then it should be fine.
